File: src/main/java/io/xol/chunkstories/api/exceptions/net/IllegalPacketException.java
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package io.xol.chunkstories.api.exceptions.net;

import io.xol.chunkstories.api.net.Packet;

public class IllegalPacketException extends Exception {

	Packet packet;

	public IllegalPacketException(Packet packet) {
		this.packet = packet;
	}

	@Override
	public String getMessage() {
		return "Illegal packet received : " + packet.getClass().getName() + "";
	}

	/**
	 * 
	 */
	private static final long serialVersionUID = 4148448942644331785L;

}

File: src/main/java/xyz/chunkstories/api/exceptions/net/IllegalPacketException.kt
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.api.exceptions.net

import xyz.chunkstories.api.net.Packet

class IllegalPacketException(internal var packet: Packet) : Exception() {

    override val message: String
        get() {
        return "Illegal packet received : " + packet.javaClass.name + ""
    }

    companion object {

        /**
         *
         */
        private val serialVersionUID = 4148448942644331785L
    }

}
